src/main/java/it/pagopa/transactions/services/v1/TransactionsService.java

```diff
@@ -1457,45 +1456,46 @@ private Mono<Boolean> wasTransactionAuthorized(
 
     @Retry(name = "addUserReceipt")
     public Mono<TransactionInfoDto> addUserReceipt(
-            String transactionId,
-            AddUserReceiptRequestDto addUserReceiptRequest
+                                                   String transactionId,
+                                                   AddUserReceiptRequestDto addUserReceiptRequest
     ) {
-        return transactionsViewRepository
-                .findById(transactionId)
+        return eventsRepository
```

Comment:
What do you think if, instead of retrieving activated event, we anticipate the reduce operation done into handler here (passing the base transaction event in the handler) using the base transaction domain object to retrieve informations such as rpt ids etc.. this way we can save a select on event store (that is, reduce operation retrieve the same event retrieved here).
This can be applied to all methods, also to authorization requested, avoiding same query to be performed twice

Reply:
Good point! I'm a bit concerned about the size of the PR, maybe it's better to create a separate optimization task dedicated to this aspect?

Reply:
Ok for me, transaction activated event and view are almost the same in size so there would not be too much difference in db load in terms of consumed RU's with this implementation.

"Reduce operation" anticipation can be posticipate to a separate PR :D